src/app/speechtotext/speechtotext.component.ts

```diff
@@ -101,6 +101,11 @@ export class SpeechtotextComponent implements OnInit {
         this.store.dispatch(new speechactions.SearchAction(false));
       }
     });
+
+    this.timer = Observable.timer(1500);
```

Comment:
there is already a timer at line 91, use that instead of using a new timer.
whenever `t==1` you can change the message to  'Listening'